tests/Unit/IO/H5/Test_CheckH5PropertiesMatch.cpp

```diff
@@ -0,0 +1,132 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <string>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/TensorData.hpp"
+#include "IO/H5/AccessType.hpp"
+#include "IO/H5/CheckH5PropertiesMatch.hpp"
+#include "IO/H5/File.hpp"
+#include "IO/H5/VolumeData.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/FileSystem.hpp"
+
+namespace {
+
+// Helper function to set up test files
+template <typename DataType>
+void setup_test_files(const std::vector<std::string>& h5_file_names) {
+  // Sample volume data
```

Comment:
I'd add a `REQUIRE(h5_file_names.size() == 3);` to prevent accidental misuse of this function :)